File: week3.1/capitalize_title.js
var capitalizeTitle = function (title) {
  // convert string to array with lower case
  const words = title.toLowerCase().split(" ");
  // define an array to track each word to capitalize
  const result = [];
  // convert each word to capitalize using for loop
  for (let word of words) {
    if (word.length > 2) {
      result.push(word[0].toUpperCase() + word.slice(1));
    } else {
      result.push(word);
    }
  }
  return result.join(" ");
};
console.log(capitalizeTitle("capiTalIze tHe titLe"));
console.log(capitalizeTitle("First leTTER of EACH Word"));
console.log(capitalizeTitle("i lOve leetcode"));
// time complexity - O(n)
// space complexity - O(n)

File: week3.1/capitalize_title.py
class Solution(object):
    def capitalizeTitle(self, title:str)->str:
        """
        :type title: str
        :rtype: str
        """
        # approch -1
        # ==============================================
        # words = title.lower().split(" ")
        # result = []
        # convert each word to capitalize
        # for word in words:
        #     if len(word)>2:
        #         result.append(word[0].upper()+word[1:])
        #     else:
        #         result.append(word)
        # return " ".join(result)
        # ==============================================
        # approch - 2
        # ==============================================
        words = title.lower().split(" ")
        result = [word[0].upper()+word[1:] if len(word)>2 else word for word in words  ]
        return " ".join(result)
        

s = Solution()
print(s.capitalizeTitle("capiTalIze tHe titLe"))
print(s.capitalizeTitle("First leTTER of EACH Word"))
print(s.capitalizeTitle("i lOve leetcode"))